tools/maven/strict-checkstyle.xml

```diff
@@ -0,0 +1,550 @@
+what the following rules do, please see the checkstyle configuration
+page at http://checkstyle.sourceforge.net/config.html.
+
+This file is based on the checkstyle file of Apache Beam.
+-->
+
+<module name="Checker">
+
+  <!--<module name="FileTabCharacter">-->
+    <!--&lt;!&ndash; Checks that there are no tab characters in the file. &ndash;&gt;-->
+  <!--</module>-->
+
+  <module name="NewlineAtEndOfFile">
+    <!-- windows can use \n\r vs \n, so enforce the most used one ie UNIx style -->
+    <property name="lineSeparator" value="lf" />
+  </module>
+
+  <module name="RegexpSingleline">
+    <!-- Checks that TODOs don't have stuff in parenthesis, e.g., username. -->
+    <property name="format" value="((//.*)|(\*.*))TODO\(" />
+    <property name="message" value="TODO comments must not include usernames." />
+    <property name="severity" value="error" />
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="format" value="\s+$"/>
+    <property name="message" value="Trailing whitespace"/>
+    <property name="severity" value="error"/>
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="format" value="Throwables.propagate\("/>
+    <property name="message" value="Throwables.propagate is deprecated"/>
+    <property name="severity" value="error"/>
+  </module>
+
+  <!-- Prevent *Tests.java as tools may not pick them up -->
+  <!--<module name="RegexpOnFilename">-->
+    <!--<property name="fileNamePattern" value=".*Tests\.java$" />-->
+  <!--</module>-->
+
+  <!-- Allow use of comment to suppress javadocstyle -->
+  <module name="SuppressionCommentFilter">
+    <property name="offCommentFormat" value="CHECKSTYLE.OFF\: ([\w\|]+)"/>
+    <property name="onCommentFormat" value="CHECKSTYLE.ON\: ([\w\|]+)"/>
+    <property name="checkFormat" value="$1"/>
+  </module>
+  <module name="SuppressionFilter">
+    <property name="file" value="${checkstyle.suppressions.file}" default="suppressions.xml" />
+  </module>
+
+  <!-- Check that every module has a package-info.java -->
+  <!--<module name="JavadocPackage"/>-->
+
+  <!--
+
+  FLINK CUSTOM CHECKS
+
+  -->
+
+  <module name="FileLength">
+    <property name="max" value="2500"/>
+  </module>
+
+  <!-- All Java AST specific tests live under TreeWalker module. -->
+  <module name="TreeWalker">
+
+    <!--
+
+    FLINK CUSTOM CHECKS
+
+    -->
+
+    <module name="RegexpSinglelineJava">
+      <property name="format" value="^\t* +\t*\S" />
+      <property name="message"
+                value="Line has leading space characters; indentation should be performed with tabs only." />
+      <property name="ignoreComments" value="true" />
+    </module>
+
+    <!-- Prohibit T.getT() methods for standard boxed types -->
+    <module name="Regexp">
+      <property name="format" value="Boolean\.getBoolean"/>
+      <property name="illegalPattern" value="true"/>
+      <property name="message" value="Use System.getProperties() to get system properties."/>
```

Comment:
The section comment is `<!-- Prohibit T.getT() methods for standard boxed types -->`.